### PR TITLE
add sqlserver basic instance data_source and resource file

### PR DESCRIPTION
**Pull request by cyberHermanwang**
--data_source
make testacc TESTARGS="-run=TestAccTencentCloudDataSqlserverBasicInstances"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudDataSqlserverBasicInstances -timeout 120m
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/gendoc	(cached) [no tests to run]
=== RUN   TestAccTencentCloudDataSqlserverBasicInstances
--- PASS: TestAccTencentCloudDataSqlserverBasicInstances (780.57s)
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud	781.489s
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/connectivity	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/internal/helper	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/ratelimit	[no test files]
--resource
 make testacc TESTARGS="-run=TestAccTencentCloudSqlserverBasicInstanceResource"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudSqlserverBasicInstanceResource -timeout 120m
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/gendoc	(cached) [no tests to run]
=== RUN   TestAccTencentCloudSqlserverBasicInstanceResource
--- PASS: TestAccTencentCloudSqlserverBasicInstanceResource (1319.44s)
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud	1320.255s
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/connectivity	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/internal/helper	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/ratelimit	[no test files]

### CODE REVIEW

File: tencentcloud/resource_tc_sqlserver_basic_instance.go [hunk 1]
```diff
@@ -0,0 +1,659 @@
+/*
+Provides a SQL Server instance resource to create Basic database instances.
```
Comment (gailwang) [inline tencentcloud/resource_tc_sqlserver_basic_instance.go:2]: basic

File: tencentcloud/resource_tc_sqlserver_basic_instance.go [hunk 2]
```diff
@@ -0,0 +1,659 @@
...
+			},
+			"charge_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      COMMON_PAYTYPE_POSTPAID,
+				ForceNew:     true,
+				ValidateFunc: validateAllowedStringValue([]string{COMMON_PAYTYPE_PREPAID, COMMON_PAYTYPE_POSTPAID}),
+				Description:  "Pay type of the SQL Server Basic instance. For now, only `POSTPAID_BY_HOUR` is valid.",
+			},
+			"goods_num": {
```
Comment (gailwang) [inline tencentcloud/resource_tc_sqlserver_basic_instance.go:95]: 这个参数不合适放在这里，如果需要 可以直接用terraform里内置的count

File: tencentcloud/resource_tc_sqlserver_basic_instance.go [hunk 3]
```diff
@@ -0,0 +1,659 @@
...
+	_ = d.Set("auto_renew", instance.RenewFlag)
+	_ = d.Set("availability_zone", instance.Zone)
+	_ = d.Set("project_id", instance.ProjectId)
+	_ = d.Set("create_time", instance.CreateTime)
+	_ = d.Set("status", instance.Status)
+	_ = d.Set("cpu", instance.Cpu)
+	_ = d.Set("memory", instance.Memory)
+	_ = d.Set("storage", instance.Storage)
+	_ = d.Set("vip", instance.Vip)
+	_ = d.Set("vport", instance.Vport)
```
Comment (gailwang) [inline tencentcloud/resource_tc_sqlserver_basic_instance.go:365]: 看起来和原来的实例 资源有大段重复，有考虑过在原来的实例上加一个basic/default的选项来整合到一起吗

File: tencentcloud/service_tencentcloud_sqlserver.go [hunk 4]
```diff
@@ -1369,3 +1384,159 @@ func (me *SqlserverService) RecycleDBInstance(ctx context.Context, instanceId st
...
+	if err != nil {
+		errRet = err
+		return
+	}
+	if response == nil || response.Response == nil {
+		errRet = fmt.Errorf("TencentCloud SDK return nil response, %s", request.GetAction())
+		return
+	}
+
+	dealId := *response.Response.DealName
```
Comment (gailwang) [inline tencentcloud/service_tencentcloud_sqlserver.go:1465]: 写了goodsnum这里如果goodsnum>1这里返回的还是一个ID吗

File: tencentcloud/service_tencentcloud_sqlserver.go [hunk 5]
```diff
@@ -1369,3 +1384,159 @@ func (me *SqlserverService) RecycleDBInstance(ctx context.Context, instanceId st
...
+
+	ratelimit.Check(request.GetAction())
+	_, err := me.client.UseSqlserverClient().UpgradeDBInstance(request)
+	if err != nil {
+		return err
+	}
+
+	//check status not expanding
+	retryCount := 0
+	errRet = resource.Retry(10*readRetryTimeout, func() *resource.RetryError {
```
Comment (gailwang) [inline tencentcloud/service_tencentcloud_sqlserver.go:1501]: 下面这段过于难理解，需要加注释

**Comment by cyberHermanwang** [comment]
make testacc TESTARGS="-run=TestAccTencentCloudSqlserverBasicInstanceResource"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudSqlserverBasicInstanceResource -timeout 120m
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/gendoc	(cached) [no tests to run]
=== RUN   TestAccTencentCloudSqlserverBasicInstanceResource
--- PASS: TestAccTencentCloudSqlserverBasicInstanceResource (647.81s)
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud	648.583s
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/connectivity	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/internal/helper	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/ratelimit	[no test files]

File: tencentcloud/resource_tc_sqlserver_basic_instance.go [hunk 6]
```diff
@@ -0,0 +1,652 @@
...
+	ctx := context.WithValue(context.TODO(), logIdKey, logId)
+	client := meta.(*TencentCloudClient).apiV3Conn
+	sqlserverService := SqlserverService{client: client}
+	tagService := TagService{client: client}
+	region := client.Region
+	payType := d.Get("charge_type").(string)
+
+	var outErr, inErr error
+	instanceId := d.Id()
+	//update name
```
Comment (gailwang) [inline tencentcloud/resource_tc_sqlserver_basic_instance.go:415]: d.Partial(true)

File: tencentcloud/resource_tc_sqlserver_basic_instance.go [hunk 7]
```diff
@@ -0,0 +1,652 @@
...
+		replaceTags, deleteTags := diffTags(oldTags.(map[string]interface{}), newTags.(map[string]interface{}))
+
+		resourceName := BuildTagResourceName("sqlserver", "instance", region, instanceId)
+		if err := tagService.ModifyTags(ctx, resourceName, replaceTags, deleteTags); err != nil {
+			return err
+		}
+
+		d.SetPartial("tags")
+	}
+
```
Comment (gailwang) [inline tencentcloud/resource_tc_sqlserver_basic_instance.go:570]: d.Partial(false)

**Review by gailwang — APPROVED** [review]

**Pull request merged**